### PR TITLE
Rewrite PCA in JASP results

### PR DESCRIPTION
**Pull request by vankesteren**
- added correlation/covariance switch (fixes [#412](jasp-stats/jasp-issues#412))
- fixed the PCA parts of [#4](jasp-stats/INTERNAL-jasp#4)
- added geomin rotations
- updated the scree plot to be more in line with the JASP theme
- updated PCA tests
- added qgraph visual test support

### CODE REVIEW

**Comment by vankesteren** [comment]
This also already has some code for exporting the principal components to the dataset, but that cannot be implemented pending further development of the code to export data from R (multiple, data-dependent number of columns)

File: JASP-Engine/JASP/R/principalcomponentanalysis.R [hunk 1]
```diff
@@ -15,850 +15,430 @@
...
-PrincipalComponentAnalysis <- function(dataset = NULL, options, perform = "run",
-																			callback = function(...) list(status="ok"), state = NULL, ...) { 
-	return(mainFunctionPCAEFA(type = "pca", dataset = NULL, options, perform = "run",
-																			callback = function(...) list(status="ok"), state = NULL, ...))
+PrincipalComponentAnalysis <- function(jaspResults, dataset, options, ...) {
+  jaspResults$addCitation("Revelle, W. (2018) psych: Procedures for Personality and Psychological Research, Northwestern University, Evanston, Illinois, USA, https://CRAN.R-project.org/package=psych Version = 1.8.12.")
+  
+  # Read dataset
+  dataset <- .pcaReadData(dataset, options)
+  ready   <- .pcaCheckErrors(dataset, options)
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/principalcomponentanalysis.R]: Nitpick: it is quite tempting to think the `ready` boolean is the outcome of the error check. This is not the case (it has nothing to do with what else happens in that function).. but to find out you have to look in the error check. Maybe do this separately?

File: JASP-Engine/JASP/R/principalcomponentanalysis.R [hunk 2]
```diff
@@ -15,850 +15,430 @@
...
-		for (j in 1:i) {
-			FactorCorrelations[["data"]][[i]][[j+1]] <- unname(dat[j])
-		}
-
-		names(FactorCorrelations[["data"]][[i]]) <- sapply(FactorCorrelations[["schema"]][["fields"]],"[[",'name')[seq_along(FactorCorrelations[["data"]][[i]])]
-	}
-	
-	return(FactorCorrelations)
+.pcaLoadingsTable <- function(modelContainer, dataset, options, ready) {
+  if (!is.null(modelContainer[["loatab"]]) || !ready) return()
```
[the remark this answers is not in the document]
Comment (vankesteren) [reply JASP-Engine/JASP/R/principalcomponentanalysis.R]: This is already there for the goodness of fit table, and for this table we don't know the number of columns ahead of time.

File: JASP-Engine/JASP/R/principalcomponentanalysis.R [hunk 3]
```diff
@@ -15,850 +15,430 @@
...
+  pcaResults <- .pcaComputeResults(modelContainer, dataset, options)
+  if (modelContainer$getError()) return()
+  
+  goftab[["model"]] <- "Model"
+  goftab[["chisq"]] <- pcaResults$STATISTIC
+  goftab[["df"]]    <- pcaResults$dof
+  goftab[["p"]]     <- pcaResults$PVAL
+  
+  if (pcaResults$dof < 0)
+    goftab$addFootnote(message = "Degrees of freedom below 0, model is unidentified.", symbol = "<em>Note.</em>")
```
[the remark this answers is not in the document]
Comment (vankesteren) [reply JASP-Engine/JASP/R/principalcomponentanalysis.R]: Fixed.

File: JASP-Engine/JASP/R/principalcomponentanalysis.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (TimKDJ) [reply JASP-Engine/JASP/R/principalcomponentanalysis.R]: You could just make a single row / col table, like it used to be before
We want the rewrite to be >= the old situation, right? ;)

File: JASP-Engine/JASP/R/principalcomponentanalysis.R (hunk 1 above)
Comment (vankesteren) [reply JASP-Engine/JASP/R/principalcomponentanalysis.R]: fixed.

**Comment by vankesteren** [comment]
I will retract and resubmit this PR after I rebasing and adding another table based on [#468](https://github.com/jasp-stats/jasp-issues/issues/468)

File: JASP-Engine/JASP/R/principalcomponentanalysis.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (vankesteren) [reply JASP-Engine/JASP/R/principalcomponentanalysis.R]: fixed.

**Pull request closed without merging**